gpu/kinfu_large_scale/tools/kinfuLS_app.cpp

```diff
@@ -977,29 +981,59 @@ struct KinFuLSApp
   }
 
   void
-  startMainLoop (bool triggered_capture)
+  startMainLoop (bool triggered_capture, bool oni2_interface = false)
   {   
-    using namespace openni_wrapper;
-    using DepthImagePtr = boost::shared_ptr<DepthImage>;
-    using ImagePtr = boost::shared_ptr<Image>;
+    boost::signals2::connection c;
 
-    std::function<void (const ImagePtr&, const DepthImagePtr&, float)> func1 = [this] (const ImagePtr& img, const DepthImagePtr& depth, float constant)
-    {
-      source_cb2 (img, depth, constant);
-    };
-    std::function<void (const DepthImagePtr&)> func2 = [this] (const DepthImagePtr& depth) { source_cb1 (depth); };
-    std::function<void (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr&) > func3 = [this] (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr& cloud)
+    if(oni2_interface)
     {
-      source_cb3 (cloud);
-    };
+      using namespace pcl::io;
+      using DepthImagePtr = boost::shared_ptr<DepthImage>;
+      using ImagePtr = boost::shared_ptr<Image>;
+
+      std::function<void (const ImagePtr&, const DepthImagePtr&, float)> func1 = [this] (const ImagePtr& img, const DepthImagePtr& depth, float constant)
+      {
+        source_cb2 (img, depth, constant);
+      };
+      std::function<void (const DepthImagePtr&)> func2 = [this] (const DepthImagePtr& depth) { source_cb1 (depth); };
+      std::function<void (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr&) > func3 = [this] (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr& cloud)
+      {
+        source_cb3 (cloud);
+      };
+
+      bool need_colors = integrate_colors_ || registration_ || enable_texture_extraction_;
 
-    bool need_colors = integrate_colors_ || registration_ || enable_texture_extraction_;
+      if ( pcd_source_ && !capture_.providesCallback<void (const pcl::PointCloud<pcl::PointXYZRGBA>::ConstPtr&)>() ) {
+        std::cout << "grabber doesn't provide pcl::PointCloud<pcl::PointXYZRGBA> callback !\n";
+      }
```

Comment:
Are similar checks for other callbacks useful? (i.e. providesCallback for func1 and func2)

Reply:
Well this check is overkilling since we are now printing exception `what()` on https://github.com/PointCloudLibrary/pcl/pull/3391/commits/f03110c98951c9ad8f03ed2a3b27ea87a6ae9769

If we tried to register a callback that doesn't exist, an exception is raised and the reason is printed in the console, so I'll remove this part